src/SPC/builder/BuilderBase.php

```diff
@@ -186,15 +200,21 @@ public function proveExts(array $extensions, bool $skip_check_deps = false): voi
         $this->emitPatchPoint('after-exts-extract');
         foreach ($extensions as $extension) {
             $class = CustomExt::getExtClass($extension);
+            /** @var Extension $ext */
             $ext = new $class($extension, $this);
             $this->addExt($ext);
+            if (in_array($extension, $shared_build_extensions)) {
+                $ext->setBuildShared();
+            } else {
+                $ext->setBuildStatic();
+            }
```

Comment:
it could be both shared and static

Reply:
If we build static, we cannot load shared one at the same time. It could be, but not necessary.